tests/lib/nested.sh

```diff
@@ -1,13 +1,69 @@
 #!/bin/bash
 
+set -e
+
+# Define where helpers are stored to support scenario when the this script is used
+# to manage vms from other projects
+SCRIPTPATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+TESTSLIB=${TESTSLIB:-$SCRIPTPATH}
+
+USE_CLOUD_INIT=${USE_CLOUD_INIT:-"true"}
+ENABLE_SECURE_BOOT=${ENABLE_SECURE_BOOT:-"true"}
+OVMF_CODE=${OVMF_CODE:-"secboot"}
+OVMF_VARS=${OVMF_VARS:-"ms"}
+EMULATE_TPM=${EMULATE_TPM:-"true"}
+BUILD_SNAPD_FROM_CURRENT=${BUILD_SNAPD_FROM_CURRENT:-"true"}
+UPDATE_PC_KERNEL=${UPDATE_PC_KERNEL:-"false"}
+CUSTOM_IMAGE_URL=${CUSTOM_IMAGE_URL:-}
+BUILD_NEW_IMAGE=${BUILD_NEW_IMAGE:-"false"}
+CORE_CHANNEL=${CORE_CHANNEL:-"edge"}
+PWD=${PWD:-$(pwd)}
+
+WORK_DIR=${WORK_DIR:-"/tmp/work-dir"}
+NESTED_VM=${NESTED_VM:-"nested-vm"}
+NESTED_SSH_PORT=${NESTED_SSH_PORT:-"8022"}
+NESTED_MON_PORT=${NESTED_MON_PORT:-"8888"}
+NESTED_ARCHITECTURE=${NESTED_ARCHITECTURE:-"amd64"}
+NESTED_TYPE=${NESTED_TYPE:-"core"}
+NESTED_SMP=${NESTED_SMP:-"1"}
+NESTED_MEM=${NESTED_MEM:-"4096"}
+
+# Define defaults for environment variables
+SPREAD_SYSTEM=${SPREAD_SYSTEM:-"$(lsb_release -is | tr '[:upper:]' '[:lower:]')-$(lsb_release -rs)-64"}
+SPREAD_BACKEND=${SPREAD_BACKEND:-"google"}
+
+show_help() {
+    echo "Nested library usage:"
+    echo ""
+    echo "1. Define the variables used to create the VM:"
+    echo "$ export BUILD_SNAPD_FROM_CURRENT=false"
+    echo "$ export UPDATE_PC_KERNEL=false"
+    echo "$ export BUILD_NEW_IMAGE=true"
+    echo "$ export EMULATE_TPM"
+    echo ""
+    echo "2. Source the nested library"
+    echo "$ . tests/lib/nested.sh"
+    echo ""
+    echo "3. Create the VM and start it as a systemd unit"
+    echo "$ create_nested_core_vm"
+    echo "$ start_nested_core_vm"
+    echo ""
+    echo "4. Check VM status"
+    echo "$ systemctl status nested-vm"
+    echo ""
+    echo "5. Connect to the VM"
+    echo "$ execute_remote"
```

Comment:
This needs args, or are they optional? Should it say `execute_remote [command]` ?